R/add_fclcm_phase.R

```diff
@@ -1,30 +1,105 @@
+#' @param fc_phase_3 The name of the value "Phase 3 FC" (by default) in the food consumption phase.
+#' @param fc_phase_4 The name of the value "Phase 4 FC" (by default) in the food consumption phase.
+#' @param fc_phase_5 The name of the value "Phase 5 FC" (by default) in the food consumption phase.
+#' @param lcs_cat_var Column name containing livelihood coping category.
+#' @param lcs_cat_none The name of the value "None" (by default) in the livelihood coping category.
+#' @param lcs_cat_stress The name of the value "Stress" (by default) in the livelihood coping category.
+#' @param lcs_cat_crisis The name of the value "Crisis" (by default) in the livelihood coping category.
+#' @param lcs_cat_emergency The name of the value "Emergency" (by default) in the livelihood coping category.
+#' @param fclcm_phase_var A character vector which will be the column name for FSLC phase.
+#' @return Returns a dataframe with a additional column for FCLC phase.
+#' @export
+#'
+#' @examples
+#' test_df <- data.frame(lcs_cat_yes = c("None", "Stress"),
+#' fc_phase = c("Phase 1 FC","Phase 2 FC"))
+#' test_df |> add_fclcm_phase()
+
+add_fclcm_phase <- function(dataset,
+                            fc_phase_var = "fc_phase",
+                            fc_phase_1 = "Phase 1 FC",
+                            fc_phase_2 = "Phase 2 FC",
+                            fc_phase_3 = "Phase 3 FC",
+                            fc_phase_4 = "Phase 4 FC",
+                            fc_phase_5 = "Phase 5 FC",
+                            lcs_cat_var = "lcs_cat_yes", # need to cross check with saeaad/ yann
+                            lcs_cat_none = "None" ,
+                            lcs_cat_stress = "Stress",
+                            lcs_cat_crisis = "Crisis",
+                            lcs_cat_emergency = "Emergency",
+                            fclcm_phase_var = "fclcm_phase"
+) {
+
+  if(sum(is.na(dataset[[fc_phase_var]])) > 0){ warning(glue::glue({{fc_phase_var}}, " has NA(s)"))}
+  if(sum(is.na(dataset[[lcs_cat_var]])) > 0){ warning(glue::glue({{lcs_cat_var}}, " has NA(s)"))}
+
+  if(fclcm_phase_var %in% names(dataset)){warning(glue::glue({{fclcm_phase_var}}, "is already existed in the dataset and will be replaced with new value"))}
+
+  fc_phase_all_phase <- c(fc_phase_1,fc_phase_2,fc_phase_3,fc_phase_4,fc_phase_5)
+  lcs_cat_all_phase <- c(lcs_cat_none,lcs_cat_stress,lcs_cat_crisis,lcs_cat_emergency)
+
+
+  if(!fc_phase_var %in% names(dataset)){
+    stop(glue::glue({{fc_phase_var}}, " column is not found in the dataset"))
+  }
+
+  if(!lcs_cat_var %in% names(dataset)){
+    stop(glue::glue({{lcs_cat_var}}, " column is not found in the dataset"))
+  }
+
+
+  if(!all(unique(na.omit(dataset[[lcs_cat_var]])) %in% lcs_cat_all_phase)){
+    print(unique(dataset[[lcs_cat_var]])[!unique(dataset[[lcs_cat_var]]) %in% lcs_cat_all_phase])
+    stop("The above value(s) are not specified/possibile wrong entry")

```

Comment:
message a bit confusing. Maybe trying to be clearer? "these values are present in the dataset but are not defined in the function arguments?